File: src/main/java/app/application/utils/LinkParameterBuilder.java
package app.application.utils;

import org.springframework.stereotype.Service;

import java.util.HashMap;
import java.util.Map;

@Service
public class LinkParameterBuilder {

	public Map<String, String> buildParameterMap(String query){
		Map<String, String> parameterMap = new HashMap<>();
		if(query != null){
			String[] params = query.split("&");
			for (String param : params){
				String[] split = param.split("=");
				parameterMap.put(split[0], split[1]);
			}
		}
		return parameterMap;
	}

}

File: src/main/java/app/application/utils/LinkParameterBuilder.kt
package app.application.utils

import org.springframework.stereotype.Service
import java.util.HashMap

@Service
class LinkParameterBuilder {

    fun buildParameterMap(query: String?): Map<String, String> {
        val parameterMap: MutableMap<String, String> = HashMap()
        if (query != null) {
            val params = query.split("&")
            for (param in params) {
                val split = param.split("=")
                parameterMap[split[0]] = split[1]
            }
        }
        return parameterMap
    }

}